app/controllers/categories_controller.rb

```diff
@@ -0,0 +1,55 @@
+class CategoriesController < ApplicationController
+  def index
+    @categories = Category.all
+  end
+
+  def new
+    @category = Category.new
+  end
+
+  def create
+    @category = Category.new(category_params)
+    if @category.save
+      flash[:success] = "#{@category.title} added!"
+      redirect_to category_path(@category)
+    else
+      flash[:failure] = "Create a category that doesn't already exist!"
+      render :new
+    end
+  end
+
+  def show
+    @category = Category.find(params[:id])
+  end
+
+  def destroy
+    category = Category.find(params[:id])
+    category.destroy
+
+    flash[:success] = "#{category.title} was successfully deleted!"
+    redirect_to categories_path
+  end
+
+  def edit
+    @category = Category.find(params[:id])
+  end
+
+  def update
+    @category= Category.find(params[:id])
```

Comment:
another shortcut could be used here...`@category = Category.update(params[:id], category_params)`